mod_app/utils/mixins.py

```diff
@@ -25,3 +28,108 @@ def link_preview(self, obj):
             '<img src="{}" style="max-width: 15rem;" />',
             obj.url,
         )
+
+
+class EmailMixin:
+    def save_model(self, request, obj, form, change):
+        super().save_model(request, obj, form, change)
+        updated_by = request.user.username
+
+        devs = [user.username for user in User.objects.filter(groups__name="devs")]
+
+        if updated_by in devs:
+            return
+
+        researchers = User.objects.filter(groups__name="Researchers").exclude(
+            username=updated_by
+        )
+        recipients = [user.email for user in researchers]
+
+        if "staging" in request.build_absolute_uri():
+            website = "staging"
+        else:
+            website = "production"
+
+        instance_admin_url = request.build_absolute_uri()
+        instance_url = request.build_absolute_uri(obj.get_absolute_url())
+        # Check if it's a new instance or an update
+        if change:
+            subject = f"A {obj.__class__.__name__} has been updated! || MOD"
+
+            html_message = render_to_string(
+                "email_template_update.html",
+                {
+                    "updated_by": updated_by,
+                    "instance": obj,
+                    "instance_admin_url": instance_admin_url,
+                    "instance_url": instance_url,
+                    "website": website,
+                },
+            )
+            send_mail(
+                subject,
+                "",
+                "notifications@museumofdreamworlds.org",
+                recipients,
+                html_message=html_message,
+                fail_silently=False,
+            )
```

Comment:
Repeated on lines 91-98, could just be written once after the if/else blocks give you subject and message?